content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,714 @@
+
+```json
+{
+  "compilerOptions": {
+    "strict": true,
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "jsx": "preserve",
+    "jsxImportSource": "solid-js",
+    "types": ["vite/client"],
+    "noEmit": true,
+    "isolatedModules": true
+  }
+}
+```
+
+4. Create a TypeScript or `.tsx` file to test the setup.
+
+```typescript
+import { type Component } from "solid-js";
+
+function MyTsComponent(): JSX.Element {
+  return (
+    <div>
+      <h1>This is a TypeScript component</h1>
+    </div>
+  );
+}
+
+export default MyTsComponent;
+```
+
+To use in an existing JavaScript component, import the TypeScript component:
+
+```javascript
+import MyTsComponent from "./MyTsComponent";
+
+function MyJsComponent() {
+  return (
+    <>
+      {/* ... */}
+      <MyTsComponent />
+    </>
+  );
+}
+```
+
+<Aside>
+**Note:** If you wish to change the entry point file from `index.jsx` to `index.tsx`, you need to modify the `src` attribute in `<script>` to look like the following:
+
+```htmlembedded
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <!-- ... -->
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <script src="/src/index.tsx" type="module"></script>
+  </body>
+</html>
+```
+</Aside>
+
+## API types
+
+Solid is written in TypeScript, meaning everything is typed out of the box.
+
+The [API Documentation](/references) details the types of all the API calls. 
+In addition, there are several helpful type definitions to make it easier when you need to specify explicit types.
+
+### Signals
+
+Using `createSignal<T>`, you can define the signal's type as `T`. 
+
+```typescript
+const [count, setCount] = createSignal<number>();
+```
+
+Here, `createSignal` has the return type `Signal<number | undefined>`, corresponding to the type passed into it, and `undefined` as it was uninitialized. 
+This resolves to a getter-setter [tuple](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types), both of which are generically typed:
+
+```typescript
+import type { Signal, Accessor, Setter } from "solid-js";
+type Signal<T> = [get: Accessor<T>, set: Setter<T>];
+```
+
+In this case, the `count` signal getter has the type `Accessor<number | undefined>`. 
+In Solid, `Accessor<T>` is equivalent to `() => T`. 
+Here, it means `count` is a function that returns either a `number` or `undefined`.
+
+The reason `undefined` is included in the type is because the signal was not initialized. 
+Therefore, its initial state can be `undefined`.
+
+The signal setter, `setCount`, is of type `Setter<number | undefined>`. 
+This complex type roughly resembles `(value? : number | ((prev?: number) => number)) => number`, accommodating both a direct number or a function which takes a potential pervious value that results in a number.
```

Comment:
```suggestion
This complex type roughly resembles `(value? : number | ((prev: number | undefined) => number | undefined)) => number | undefined`, accommodating both a direct number or a function which takes a potential previous value that results in a number.
```
The function form will always be called with one argument, which you can choose to ignore; it isn't correct to say that parameter is optional.
Also, the return type of the setter will match the type passed to it, behaviour matching a normal assignment. Not sure if that's worth documenting though.